packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -84,6 +84,10 @@ contract TaikoL1 is
      *          will be the first transaction in the block -- if there are
      *          n transactions in `txList`, then there will be up to n+1
      *          transactions in the L2 block.
+     *
+     *        - inputs[2] a byte32 that determins which verifier to run.
+     *
+     *        - inputs[3] a ZKP that verifies all signatures.
```

Comment:
Maybe good to add here that it also validates the RLP encoding of the transaction (and also things like data compression) which would be the reason invalid blocks cannot happen anymore.